packages/core/src/domain/conversion/bundle-modifications/__tests__/post-process-bundle.test.ts

```diff
@@ -0,0 +1,225 @@
+
+      const existingExtension = {
+        url: "http://example.com/existing",
+        valueString: "test",
+      };
+      condition.extension = [existingExtension];
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.addExtensionToConversion(bundle, documentExtension);
+      const updCondition = updatedBundle.entry?.[0]?.resource as Condition | undefined;
+      expect(updCondition).toBeDefined();
+      if (!updCondition) throw new Error("Bundle entry got removed!");
+
+      expect(updCondition.extension).toHaveLength(2);
+      expect(updCondition.extension).toEqual(
+        expect.arrayContaining([existingExtension, documentExtension])
+      );
+      expect(condition.extension).toHaveLength(1);
+    });
+
+    it("returns bundle unchanged when no entries exist", () => {
+      const { documentExtension } = initTest();
+      const bundle = makeBundle({ entries: [], type: "collection" });
+      const updatedBundle = bundleMods.addExtensionToConversion(bundle, documentExtension);
+      expect(updatedBundle).toEqual(bundle);
+    });
+  });
+
+  describe("addMissingRequests", () => {
+    it("appends a PUT request to each BundleEntry with correct resource references", () => {
+      const { patientId, condition } = initTest();
+      const condition2 = makeCondition({ id: faker.string.uuid() }, patientId);
+      const bundle = makeBundle({ entries: [condition, condition2], type: "collection" });
+
+      const updatedBundle = bundleMods.addMissingRequests(bundle);
+      expect(updatedBundle.entry).toHaveLength(2);
+
+      updatedBundle.entry?.forEach((entry, index) => {
+        const resource = bundle.entry?.[index]?.resource;
+        expect(entry.request).toEqual({
+          method: "PUT",
+          url: `${resource?.resourceType}/${resource?.id}`,
+        });
+      });
+    });
+
+    it("returns bundle unchanged when no entries exist", () => {
+      const bundle = makeBundle({ entries: [], type: "collection" });
+      const updatedBundle = bundleMods.addMissingRequests(bundle);
+      expect(updatedBundle).toEqual(bundle);
+    });
+  });
+
+  describe("removePatientFromConversion", () => {
+    it("throws error when multiple Patient resources exist", () => {
+      const { patientId, condition } = initTest();
+      const patient1: Patient = {
+        resourceType: "Patient",
+        id: patientId,
+        name: [{ given: ["John"], family: "Doe" }],
+      };
+      const patient2: Patient = {
+        resourceType: "Patient",
+        id: faker.string.uuid(),
+        name: [{ given: ["Jane"], family: "Doe" }],
+      };
+      const bundle = makeBundle({ entries: [condition, patient1, patient2], type: "collection" });
+
+      expect(() => bundleMods.removePatientFromConversion(bundle)).toThrow(
+        "Multiple Patient resources found in Bundle"
+      );
+      expect(bundle.entry).toHaveLength(3);
```

Comment:
ditto: testing the params or the execution of the main function? Here and everywhere else this applies, please

Reply:
removed in this case